### DISCUSSION TITLE
How we can deploy Semantic Kernel with Collaboration along Prompt Flow Using Azure Ai Foundry?

### DISCUSSION THREAD

**Discussion by lovedeepatsgit** [microsoft/semantic-kernel#10007] [Q&A] [answered]
I have successfully made a SK which uses multi-agents ( Agentic- Framework) using python, while there is official Documentation to Evaluate SK using Prompt Flow (https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/how-to-evaluate-semantic-kernel?view=azureml-api-2), from this documentation I got an Idea to deploy SK using Prompt flow, but in my code I am using "kernel_functions" which are supported in Semantic-kernel == 1.16 and for this version of SK, I need version of python > 3.10, And, on Azure Ai Foundry's prompt flow there is an environment provided by azure of latest version (mcr.microsoft.com/azureml/promptflow/promptflow-runtime:latest), but the container created from this image is of version 3.9.16, I have tried to change the version of python by making a custom image with python version 3.10.1, but while doing this some unexpected issues and error has occurred, like readiness and liveness of the container. So, according to you what is the best for deployment of Semantic-Kernels? or how should I use this custom image? or if in early 2025 Microsoft is thinking to make SK available for GA, then is Microsoft is also updating this base images which can suitable for SK's latest versions?

**Comment by sophialagerkranspandey** [comment]
@TaoChenOSU could you please take a look at this? Thanks!

**Comment by RicardoNiepel** [comment]
Referencing: Adds to this one #10022

**Comment by TaoChenOSU** [comment, marked as answer]
Hi @lovedeepatsgit,

Thank you for reporting the issue!

To use a different Python version in a custom PF image, please follow the guide provided here: https://github.com/microsoft/promptflow/issues/3344

Let us know if you see further issues.
